lib/money/money.rb

```diff
@@ -8,6 +8,16 @@ class Money
 
   class << self
     alias_method :from_amount, :new
+
+    def new(value = 0, _currency = nil)
+      if value == 0
+        @empty ||= from_amount(0).freeze
```

Comment:
`_curency` is currently a stub to provide compatibility with https://github.com/RubyMoney/money because we want to migrate to it.

So it doesn't really matter right now. But yes if we want to keep that perf patch once the transition is done, then we'll have to rethink it a bit, even though `0` in any currency is generally equal to 0 in any other currency.